### PR TITLE
Add @laravel 5 support.

### PR DESCRIPTION
**Pull request by vinkla**
I've added the @GrahamCampbell Laravel Manager to handle multiple connections. Though, I'm not sure about the Segment I'm currently returning. @jbrooksuk  maybe you know this? I initialize both `Segment` and `Segment_Client` instances but am only returning the later. Is this expected behavior? 

If you guys okay everything I can start adding better documentation as well.

``` php
/**
 * Make a new Segment client.
 *
 * @param array $config
 * @return Segment_Client
 */
public function make(array $config)
{
    $config = $this->getConfig($config);

    Segment::init(
        $config['secret'],
        $config['options']
    );

    return new Segment_Client(
        $config['secret'],
        $config['options']
    );
}
```

### CODE REVIEW

**Comment by vinkla** [comment]
@jbrooksuk the wonkiness is fixed.

**Comment by GrahamCampbell** [comment]
>  Laravel Manager to handle multiple connections.

That doesn't work. We can't have multiple connections due to the shit static design of the segment package.

File: src/Factories/SegmentFactory.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	 * Make a new Segment client.
+	 *
+	 * @param array $config
+	 * @return Segment_Client
+	 */
+	public function make(array $config)
+	{
+		$config = $this->getConfig($config);
+
+		Segment::init(
```
Comment (GrahamCampbell) [inline src/Factories/SegmentFactory.php:20]: You can't do this This will break all existing clients.

File: tests/SegmentManagerTest.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace CachetHQ\Tests\Segment;
+
+use GrahamCampbell\TestBench\AbstractTestCase as AbstractTestBenchTestCase;
+use Mockery;
+use CachetHQ\Segment\SegmentManager;
+
+class SegmentManagerTest extends AbstractTestBenchTestCase
```
Comment (GrahamCampbell) [inline tests/SegmentManagerTest.php:9]: You can't copy tests out of my existing packages without including the original license and a link back.

Comment (vinkla) [inline tests/SegmentManagerTest.php]: Okay. Send me what you want to add.

File: src/Factories/SegmentFactory.php (hunk 1 above)
Comment (vinkla) [inline src/Factories/SegmentFactory.php:20]: Yeah. Due to the Segement package setup, we'll have to drop the manager all together and just initialize the Segment init method.

**Comment by vinkla** [comment]
Closing this and starting fresh.

File: src/Factories/SegmentFactory.php (hunk 1 above)
Comment (GrahamCampbell) [inline src/Factories/SegmentFactory.php:20]: That's why I didn't get us to use it in the first place. :)

Comment (vinkla) [inline src/Factories/SegmentFactory.php:20]: Yeah, I've tried to create a second version but its hard. The Segment PHP package really needs a big update. I have tried refactoring it in https://github.com/segmentio/analytics-php/pull/46 Though, the tests keep failing.

**Pull request closed without merging**